src/tools/pulse/generateInsightBrief/generatePulseInsightBriefTool.test.ts

```diff
@@ -0,0 +1,200 @@
+                datasource: {
+                  id: 'A6FC3C9F-4F40-4906-8DB0-AC70C5FB5A11',
+                },
+                basic_specification: {
+                  measure: {
+                    field: 'Sales',
+                    aggregation: 'AGGREGATION_SUM' as const,
+                  },
+                  time_dimension: {
+                    field: 'Order Date',
+                  },
+                  filters: [],
+                },
+                is_running_total: false,
+              },
+              metric_specification: {
+                filters: [],
+                measurement_period: {
+                  granularity: 'GRANULARITY_BY_MONTH' as const,
+                  range: 'RANGE_CURRENT_PARTIAL' as const,
+                },
+                comparison: {
+                  comparison: 'TIME_COMPARISON_PREVIOUS_PERIOD' as const,
+                },
+              },
+              extension_options: {
+                allowed_dimensions: [],
+                allowed_granularities: [],
+                offset_from_today: 0,
+              },
+              representation_options: {
+                type: 'NUMBER_FORMAT_TYPE_NUMBER' as const,
+                number_units: {
+                  singular_noun: 'dollar',
+                  plural_noun: 'dollars',
+                },
+                sentiment_type: 'SENTIMENT_TYPE_NONE' as const,
+                row_level_id_field: {
+                  identifier_col: 'Order ID',
+                },
+                row_level_entity_names: {
+                  entity_name_singular: 'Order',
+                  entity_name_plural: 'Orders',
+                },
+                row_level_name_field: {
+                  name_col: 'Order Name',
+                },
+                currency_code: 'CURRENCY_CODE_USD' as const,
+              },
+              insights_options: {
+                settings: [],
+              },
+            },
+          },
+        ],
+        metric_group_context_resolved: true,
+      },
+    ],
+    now: '2025-11-15 00:00:00',
+    time_zone: 'UTC',
+  };
+
+  const mockBriefResponse = {
+    markup: 'Your sales metric shows strong performance this month with a 15% increase.',
+    generation_id: 'test-generation-id',
+    source_insights: [
+      {
+        type: 'trend',
+        markup: 'Sales have been trending upward consistently',
+      },
+    ],
+    follow_up_questions: [
+      { content: 'What factors contributed to the increase?' },
+      { content: 'How does this compare to last year?' },
+    ],
+    group_context: briefRequest.messages[0].metric_group_context,
+    not_enough_information: false,
+  };
+
+  beforeEach(() => {
+    vi.clearAllMocks();
+    resetResourceAccessCheckerSingleton();
+    mocks.mockGetConfig.mockReturnValue({
+      disableMetadataApiRequests: false,
+      boundedContext: {
+        projectIds: null,
+        datasourceIds: null,
+        workbookIds: null,
+      },
+    });
```

Comment:
Same, I feel like having the mock config may be unnecessary

Reply:
The resource access checker singleton caches the results of whether a datasource is allowed or not. Resetting it clears the cache